### ISSUE TITLE
Clean up MCA var de-registration of common components

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#11680]
Per discussion at https://github.com/open-mpi/ompi/pull/11640#discussion_r1193925478, the de-registration of MCA vars of common components is not handled well, leaving components to do somewhat ugly things to avoid segv's (e.g., malloc space for string values).

We should clean this up so that it is not unsafe, for example, to have string MCA vars with static storage in common components that will not segv during finalize.

**Comment by hjelmn** [comment]
Yeah. common was always a weird part of the MCA. At the time we implemented MPI-T support all common registration got hooked to another framework's component (btl/blah as an example). Was not the cleanest solution for sure.
